Document volume expansion and volume claim immutability (#3797)
elastic/cloud-on-k8s: docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc

ECK automatically deletes PersistentVolumeClaim resources if they are not required for any Elasticsearch node. The corresponding PersistentVolume may be preserved, depending on the configured link:https://kubernetes.io/docs/concepts/storage/storage-classes/#reclaim-policy[storage class reclaim policy].

[float]
== Updating the volume claim settings

> If the storage class allows link:https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/[volume expansion], you can increase the storage requests size in the volumeClaimTemplates. ECK will update the existing PersistentVolumeClaims accordingly, and recreate the StatefulSet automatically. If the volume driver supports `ExpandInUsePersistentVolumes`, the filesystem will be resized online, and does not require restarting the Elasticsearch process nor re-creating the Pods. If the volume driver does not support `ExpandInUsePersistentVolumes`, it is necessary to manually delete the Pods after the resize, to be recreated automatically with the expanded filesystem.

Reviewer: ```suggestion
If the storage class allows link:https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/[volume expansion], you can increase the storage requests size in the volumeClaimTemplates. ECK will update the existing PersistentVolumeClaims accordingly, and recreate the StatefulSet automatically. If the volume driver supports `ExpandInUsePersistentVolumes`, the filesystem is resized online, without the need of restarting the Elasticsearch process, or re-creating the Pods. If the volume driver does not support `ExpandInUsePersistentVolumes`, Pods must be manually deleted after the resize, to be recreated automatically with the expanded filesystem.
```

> Any other changes in the volumeClaimTemplates, such as changing the storage class or decreasing the volume size, is forbidden. In order to achieve those, you can create a new nodeSet with different settings, and remove the existing nodeSet. In practice, that's equivalent to renaming the existing nodeSet while modifying its claim settings in a single update. ECK ensures data stored on Pods of the deleted nodeSet is migrated to other nodes before removing Pods of the deleted nodeSet.

Reviewer: ```suggestion
Any other changes are forbidden in the volumeClaimTemplates, such as changing the storage class or decreasing the volume size. To make these changes, you can create a new nodeSet with different settings, and remove the existing nodeSet. In practice, that's equivalent to renaming the existing nodeSet while modifying its claim settings in a single update. Before removing Pods of the deleted nodeSet, ECK makes sure that data is migrated to other nodes.
```
